### PR TITLE
CBL-324: fix: support source build

### PR DESCRIPTION
**Pull request by jayahariv**
* support source for android was not building with ICU
* create section for including the directory and pass the argument
* complaining about arc4random and strlcat, which resolved by including linux support configs.

### CODE REVIEW

File: cmake/platform_android.cmake [hunk 1]
```diff
@@ -59,7 +59,7 @@ function(setup_litecore_build)
     )
 
     target_link_libraries(
-        LiteCore PRIVATE 
+        LiteCore PRIVATE
```
Comment (jayahariv) [inline cmake/platform_android.cmake]:
```suggestion
        LiteCore PRIVATE 
```

File: cmake/platform_android.cmake [hunk 2]
```diff
@@ -47,7 +47,7 @@ endfunction()
 function(setup_litecore_build)
     setup_litecore_build_linux()
 
-    
+
```
Comment (jayahariv) [inline cmake/platform_android.cmake]:
```suggestion
    
```

File: cmake/platform_android.cmake [hunk 3]
```diff
@@ -70,6 +70,18 @@ function(setup_litecore_build)
     )
 endfunction()
 
+function(setup_support_build)
+    setup_support_build_linux()
+    target_compile_definitions(
+        Support PRIVATE
+        -DLITECORE_USES_ICU=1
```
Comment (borrrden) [inline cmake/platform_android.cmake]: If this is needed in Android then it's needed in Linux as well, so better to move this compile definition into `platform_linux` (which both Linux and Android derive from)

File: cmake/platform_linux.cmake [hunk 4]
```diff
@@ -53,9 +53,13 @@ function(setup_litecore_build_linux)
     )
 endfunction()
 
-function(setup_support_build)
+function(setup_support_build_linux)
```
Comment (borrrden) [inline cmake/platform_linux.cmake]: If you rename this then you need to add `setup_support_build` to `platform_linux_desktop` which just calls this method.

**Comment by pasin** [comment]
@bmeike Maybe this also fixes the swedish collator issue in one of the unit tests?

File: cmake/platform_linux.cmake [hunk 5]
```diff
@@ -53,9 +53,13 @@ function(setup_litecore_build_linux)
...
     target_include_directories(
         Support PRIVATE
         LiteCore/Unix
     )
+endfunction()
+
+function(setup_support_build)
```
Comment (borrrden) [inline cmake/platform_linux.cmake]: This will conflict with the Android version in `platform_android`

File: cmake/platform_linux_desktop.cmake [hunk 6]
```diff
@@ -103,13 +103,17 @@ function(setup_litecore_build)
     endif()
 
     target_link_libraries(
-        LiteCore PRIVATE 
-        z 
-        ${ICU4C_COMMON} 
+        LiteCore PRIVATE
```
Comment (jayahariv) [inline cmake/platform_linux_desktop.cmake]:
```suggestion
        LiteCore PRIVATE 
```

File: cmake/platform_linux_desktop.cmake [hunk 7]
```diff
@@ -103,13 +103,17 @@ function(setup_litecore_build)
...
+        z
```
Comment (jayahariv) [inline cmake/platform_linux_desktop.cmake]:
```suggestion
        z 
```

File: cmake/platform_linux_desktop.cmake [hunk 8]
```diff
@@ -103,13 +103,17 @@ function(setup_litecore_build)
...
+        ${ICU4C_COMMON}
```
Comment (jayahariv) [inline cmake/platform_linux_desktop.cmake]:
```suggestion
        ${ICU4C_COMMON} 
```

**Comment by bmeike** [comment]
@pasin That would be cool!  I'll check, when this becomes available.

**Pull request merged**